WalletWasabi/Affiliation/AffiliationManager.cs

```diff
@@ -0,0 +1,51 @@
+using System.Linq;
+using System.Collections.Immutable;
+using WalletWasabi.Affiliation.Models;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net.Http;
+using WalletWasabi.Tor.Http;
+using Microsoft.Extensions.Hosting;
+using System.Collections.Generic;
+
+namespace WalletWasabi.Affiliation;
+
+public class AffiliationManager : BackgroundService, IAffiliationManager
+{
+	public AffiliationManager(Arena arena, ImmutableDictionary<AffiliationFlag, string> urls, string privateKeyHex)
+	{
+		Arena = arena;
+		HttpClient httpclient = new();
```

Comment:
Partial fix is in https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/6cbfa6ec17b955f718aa2ba8efe18b51bd16c4f3.

I briefly read the following articles:
  * https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0#consumption-patterns
  * https://www.aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/
  * https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

I think I understand the concept of `IHttpClientFactory` but I don't fully understand how [dependency injection](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-7.0) works and I need your help here. I don't know how to easily retrieve an instance of `IHttpClientFactory`.

Reply:
Do you have any opinion or advice on this?

Reply:
Regarding `HttpClient`, see https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/0121ce1c1882d9c7a280e96686fb0912ff1deae1. Nevertheless, I'm still not happy with the solution. [Documentation](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests) says:
> Another issue that developers run into is when using a shared instance of HttpClient in long-running processes. In a situation where the HttpClient is instantiated as a singleton or a static object, it fails to handle the DNS changes as described in this [issue](https://github.com/dotnet/runtime/issues/18348) of the dotnet/runtime GitHub repository.
>
> ...
>
> To address the issues mentioned above and to make HttpClient instances manageable, .NET Core 2.1 introduced two approaches, one of them being [IHttpClientFactory](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory). It's an interface that's used to configure and create HttpClient instances in an app through Dependency Injection (DI). It also provides extensions for Polly-based middleware to take advantage of delegating handlers in HttpClient.
>
> The alternative is to use SocketsHttpHandler with configured PooledConnectionLifetime. This approach is applied to long-lived, static or singleton HttpClient instances. To learn more about different strategies, see [HttpClient guidelines for .NET](https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines).

So it still seems to me that we are mixing two approaches together.

Reply:
> So what about https://github.com/kiminuo/WalletWasabi/commit/54c70d2fe1ca9ad96b6b7247c8c1fd761d2ee488 ?

This is almost the same as it was before Lucas opened this thread (see https://github.com/zkSNACKs/WalletWasabi/pull/9769#discussion_r1046316807). With the exception that i didn't use `IHttpClientFactory` to create `HttpClient`. But that doesn't make much difference since `AffiliateManager` is initialized only once.

Reply:
I still think that we are mixing two approaches but if you are ok with it I'm ok with it as well.